cli/src/pcluster/models/cluster.py

```diff
@@ -755,13 +754,21 @@ def has_running_capacity(self, updated_value: bool = False) -> bool:
                 )
         return self.__has_running_capacity
 
-    def has_running_login_nodes(self, updated_value: bool = False) -> bool:
-        """Return True if the cluster has running login nodes. Note: the value will be cached."""
+    def has_running_login_nodes(self, updated_value: bool = False, pool_name: str = None) -> bool:
+        """
+        Return True if the cluster has running login nodes, or a specific pool if a pool name is provided.
+
+        Note: the value will be cached.
+        """
```

Comment:
Nitpick: Since `get_healthy_nodes` already checks in pool_name in None, you don't need to check it here. You could just call `get_healthy_nodes(pool_name)` regardless if pool_name = None.

Reply:
Nice, thank you!